src/Raven.Server/ServerWide/Commands/CompareExchangeCommands.cs

```diff
@@ -190,12 +191,14 @@ object IBlittableResultCommand.WriteResult(object result)
         {
             var compareExchangeResult =  result switch
             {
-                CompareExchangeResult obj => obj,
+                CompareExchangeResult obj => new CompareExchangeResult { Index = obj.Index, Value = obj.Value },
```

Comment:
add copying ctor to the CompareExchangResult, we will add one field in the future and it will be hard to find all those places to fix